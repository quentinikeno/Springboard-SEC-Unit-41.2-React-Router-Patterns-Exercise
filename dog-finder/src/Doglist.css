.Doglist-list {
	display: grid;
	grid-template-columns: repeat(2, 1fr);
	gap: 10px;
	grid-auto-rows: minmax(100px, auto);
}

@media (max-width: 1200px) {
	.Doglist-list {
		display: block;
	}
}
