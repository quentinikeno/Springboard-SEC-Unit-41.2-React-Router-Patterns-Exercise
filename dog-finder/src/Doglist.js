import DogLink from "./DogLink";
import "./Doglist.css";

const Doglist = ({ dogs }) => {
	const allDogs = dogs.map((dog) => (
		<DogLink name={dog.name} src={dog.src} key={dog.name} />
	));
	return (
		<div className="Doglist">
			<h1>React Dog Finder</h1>
			<div className="Doglist-list">{allDogs}</div>
		</div>
	);
};

export default Doglist;
