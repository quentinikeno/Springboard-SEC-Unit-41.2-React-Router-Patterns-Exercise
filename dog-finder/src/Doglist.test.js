import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Doglist from "./Doglist";
import dogs from "./dogs";

it("should render without crashing", () => {
	render(
		<MemoryRouter>
			<Doglist dogs={dogs} />
		</MemoryRouter>
	);
});

it("should matchSnapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<Doglist dogs={dogs} />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("should have all the dogs in the document", () => {
	const { getByText } = render(
		<MemoryRouter>
			<Doglist dogs={dogs} />
		</MemoryRouter>
	);
	expect(getByText("Whiskey")).toBeInTheDocument();
	expect(getByText("Duke")).toBeInTheDocument();
	expect(getByText("Perry")).toBeInTheDocument();
	expect(getByText("Tubby")).toBeInTheDocument();
});
