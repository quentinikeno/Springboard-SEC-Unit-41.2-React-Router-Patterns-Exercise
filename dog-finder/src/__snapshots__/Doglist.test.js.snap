// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="Doglist"
  >
    <h1>
      React Dog Finder
    </h1>
    <div
      class="Doglist-list"
    >
      <div
        class="DogLink"
      >
        <a
          href="/dogs/whiskey"
        >
          <h3>
            Whiskey
          </h3>
          <img
            alt="Whiskey"
            src="whiskey.jpg"
          />
        </a>
      </div>
      <div
        class="DogLink"
      >
        <a
          href="/dogs/duke"
        >
          <h3>
            Duke
          </h3>
          <img
            alt="Duke"
            src="duke.jpg"
          />
        </a>
      </div>
      <div
        class="DogLink"
      >
        <a
          href="/dogs/perry"
        >
          <h3>
            Perry
          </h3>
          <img
            alt="Perry"
            src="perry.jpg"
          />
        </a>
      </div>
      <div
        class="DogLink"
      >
        <a
          href="/dogs/tubby"
        >
          <h3>
            Tubby
          </h3>
          <img
            alt="Tubby"
            src="tubby.jpg"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
